src/base/lib/mapping/mapmach.c

```diff
@@ -0,0 +1,123 @@
+/*
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+/*
+ * Purpose: memory mapping library, Mach backend for macOS
+ *
+ * Authors: Bart Oldeman.
+ */
+
+#include <string.h>
+#include <sys/mman.h>
+
+#include <mach/mach_init.h>
+#include <mach/mach_vm.h>
+
+#include "dosemu_debug.h"
+#include "mapping.h"
+
+/* ------------------------------------------------------------ */
+
+static void *alias_mapping_mach(int cap, void *target, size_t mapsize, int protect, void *source)
+{
+  mach_vm_address_t targetaddr;
+  vm_prot_t cur, max;
+  int flags;
+
+  if (target != (void *)-1) {
+    flags = VM_FLAGS_FIXED | VM_FLAGS_OVERWRITE;
```

Comment:
Should we support MAPPING_NOOVERLAP here btw?

Reply:
I don't think we need to, it's not supported by mapfile.c either, and not used anywhere for aliases.